dms/models/dms_security_mixin.py

```diff
@@ -135,6 +136,22 @@ def _apply_access_groups(self, query, mode="read"):
             )
         query.where_clause += [where_clause]
         query.where_clause_params += [self.env.user.id]
+        # Add _get_directory_ids_with_res_model_without_access
+        if self.env.context.get("use_res_model_without_access", True):
+            custom_ids = self._get_directory_ids_with_res_model_without_access(mode)
+            if len(custom_ids) > 0:
+                field = "id"
+                if self._name == "dms.file":
+                    field = "directory_id"
+                extra_clause = """
+                    "{table}".{field} NOT IN ({ids})
+                """.format(
+                    table=self._table,
+                    field=field,
+                    ids=", ".join(["%s"] * len(custom_ids)) % tuple(custom_ids),
```

Comment:
I still think we should leverage `psycopg2` placeholder expansion, which protects from injections:

```suggestion
                    ids=", ".join(["%s"] * len(custom_ids)),
```

I'm not sure if you did it this way due to something I'm missing, or if it was just a confusion... If I'm missing something please tell me 😅

Reply:
Sorry, your suggestion doesn't work because generate empty list.